src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -84,12 +117,16 @@ public Delta(Type structuralType, IEnumerable<string> updatableProperties)
         /// names, including those of dynamic properties, are ignored.</param>
         /// <param name="dynamicDictionaryPropertyInfo">The property info that is used as dictionary of dynamic
         /// properties. <c>null</c> means this entity type is not open.</param>
+        /// <param name="edmType">Edm type of the TStructuralType</param>
         public Delta(Type structuralType, IEnumerable<string> updatableProperties,
-            PropertyInfo dynamicDictionaryPropertyInfo)
+            PropertyInfo dynamicDictionaryPropertyInfo, IEdmTypeReference edmType)
         {
             _dynamicDictionaryPropertyinfo = dynamicDictionaryPropertyInfo;
             Reset(structuralType);
             InitializeProperties(updatableProperties);
+            _edmType = edmType;
+            TransientInstanceAnnotationContainer = new ODataInstanceAnnotationContainer();
+            PersistentInstanceAnnotationsContainer = new ODataInstanceAnnotationContainer();
```

Comment:
if TStructrualType supports instance annotations we should be storing persistent instance annotations there, not on the Delta<TStructuralType>